### PR TITLE
DAOS-2166 doc: add object module's readme

### PR DESCRIPTION
**Pull request by liuxuezhao**
Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: src/object/README.md [hunk 1]
```diff
@@ -1,261 +1,145 @@
...
 
-The DAOS object schema describes the definitions for object types, data protection methods, and data distribution strategies. Based on these definitions and the requirements, upper layer stacks of DAOS-SR can register a number of object classes. An object class has a unique class ID, which is a 16-bit value, and can represent a category of objects that use the same schema and schema attributes. A successfully registered object class is stored as container metadata; it is valid in the lifetime of the container. In addition, DAOS-SR also provides some pre-defined object class for the most common use.
+To avoid scaling problems and overhead common to traditional storage stack,
+DAOS objects are intentionally very simple. No default object metadata beyond
+the type and schema (object class ownership) are provided. This means that the
+system does not maintain time, size, owner, permissions and opener tracking
+attributes. All object operations are idempotent and can be processed multiple
+times with the same results. This guarantees that any operations can be repeated
+until successful or abandoned, without having to handle execute-once semantic
+and reply reconstruction.
```
Comment (wangdi1) [inline src/object/README.md]: I am not sure this idempotent thing is correct anymore, especially with the new transaction model and EC.

File: src/object/README.md [hunk 2]
```diff
@@ -1,261 +1,145 @@
...
+  to release the main IO service extream from IO request sending and reply
+  receiving (see shard_req_forward).
+- then serve the IO request locally
+- wait the forwarded IO's completion and reply client IO request.
+
+In this mode the conflict writes can be detected and serialized by the leader
+shard server. Now both modes are supported by DAOS, it can be dynamically
+configured by environment variable "DAOS_IO_SRV_DISPATCH" before loading DAOS
+server. By default DAOS works in server replication mode, and if the ENV set as
+zero then will work in client replication mode.
```
Comment (wangdi1) [inline src/object/README.md]: I did not see retry I/O is mentioned somewhere? that is a important part of src/object/

File: src/object/README.md [hunk 3]
```diff
@@ -1,261 +1,145 @@
...
+- If DAOS cannot capture the failure, for example, the DAOS client itself
+  crashed before successfully updating all replicas so some replicas may lag
+  behind. Based on the current replication protocol, the DAOS servers cannot
+  detect the missing I/O requests, so DAOS cannot guarantee data consistency
+  between replicas. The upper layer stack has to either re-submit the
+  interrupted I/O requests to enforce data consistency between replicas, or
+  abort the epoch and rollback to the consistent status of the container.
+  For example, if an application linked to the DAOS client library crashed, then
+  it should either replay uncompleted I/O requests if it has log for those I/Os,
+  or abort the uncommitted epoch.
```
Comment (wangdi1) [inline src/object/README.md]: These reasoning probably can be simplified, just say  " To keep the consistency of each replicas,  another I/O mode ...."

**Comment by wangdi1** [comment]
did not see retry I/O is mentioned somewhere? that is a important part of src/object/

**Comment by jolivier23** [comment]
can you resolve the conflicts?

**Pull request merged**